### ISSUE TITLE
Why don't you guys add a selection to delete the swp file directly?

### ISSUE THREAD

**Issue by yingshaoxo** [vim/vim#12035]
It's been many years, whenever the ssh connection gets lost, if we go back to the file, it will hit the swp error

Why do we have to do the stupid typing again to delete that file? `rm *.swp`

Why don't you guys add an option to delete it directly by selecting and hitting an enter key?

**Comment by cero1988** [comment]
the option to delete the swapfile has already been added, if you compile vim from the source code on github you con try it

**Comment by chrisbra** [comment]
that option is there only, if Vim cannot find that  process-id from the swapfile. 

I think this is a safe choice. If you want a custom behaviour, you can use a `SwapExists` autocommand.  

But perhaps the "Delete" choice can be made available always, independently of whether Vim found that process-id? Not sure if this is a good idea in general.

**Comment by brammool** [comment]
Vim is careful not to make it too easy to delete the .swp file, because it might be the only way to restore the text that you last worked on.  If you delete the .swp file it may be lost forever.
If the process is still running, it would need to be killed (gently), so that it saves all information in the .swp file, before trying to recover the text.

Unfortunately, ssh tends to drop the connection quickly.  When traveling I often use tmux to avoid trouble.  It would be much better if ssh would be able to recover the connection.

**Comment by yingshaoxo** [comment]
Why don't you guys add a confirm box, like:

Do you want to delete it? It can't be restored. (Y/N)

Y

**Comment by Shane-XB-Qian** [comment]
![image](https://user-images.githubusercontent.com/13550529/220535508-443a9839-095a-4d89-a550-25b3cd8546d3.png)

**Comment by yingshaoxo** [comment]
> ![image](https://user-images.githubusercontent.com/13550529/220535508-443a9839-095a-4d89-a550-25b3cd8546d3.png)

That looks fine to me.

**Comment by Shane-XB-Qian** [comment]
this *smart dialog had existed many years.

--
shane.xb.qian

**Comment by chrisbra** [comment]
> If the process is still running, it would need to be killed (gently), so that it saves all information in the .swp file, before trying to recover the text.

If the process is still running and it hasn't flushed the data to the swap file on the disk, then the `Recover` option is as dangerous as the `Delete` option, no?

I would just always show the delete option and leave it to the user to be sensible. If he deletes the swap-file, well so be it. On Linux, you could probably restore the swapfile from tracing the links in `/proc/<pid>/fd` and on windows, I believe you cannot delete files that are still open so it shouldn't be an issue there?

**Comment by brammool** [comment]
> > If the process is still running, it would need to be killed
> > (gently), so that it saves all information in the .swp file, before
> > trying to recover the text.
> 
> If the process is still running and it hasn't flushed the data to the
> swap file on the disk, then the `Recover` option is as dangerous as
> the `Delete` option, no?

No, it would just end up with a slightly older version of the file.
Nothing gets deleted, only when you overwrite the original file you
might loose some data.

> I would just always show the delete option and leave it to the user to
> be sensible. If he deletes the swap-file, well so be it. On Linux, you
> could probably restore the swapfile from tracing the links in
> `/proc/<pid>/fd` and on windows, I believe you cannot delete files
> that are still open so it shouldn't be an issue there? 

I doubt users who have their computer misbehave will first read the
documentation and fully understand the situation and what is the best
way to deal with it.  Vim should help the user make it easy to pick the
right choice.

If the process is still running, it might be possible to use it.  E.g.
when using tmux.  Or perhaps this process was already running longer ago
and the user just forgot about it.

Everybody knows how to delete a file, only the location and name might
be a mystery.  Adding the "D" option makes it easier to do the wrong
thing.  Only in very few situations it is useful.

-- 
An error has occurred.  Hit any user to continue.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
